cmd/ttn-lw-stack/commands/start.go

```diff
@@ -45,231 +45,229 @@ import (
+		}
+		startDefault := len(args) == 0
+		for _, arg := range args {
+			switch strings.ToLower(arg) {
+			case "is", "identityserver":
+				start.IdentityServer = true
+				start.DeviceTemplateConverter = true
+				start.QRCodeGenerator = true
+			case "gs", "gatewayserver":
+				start.GatewayServer = true
+			case "ns", "networkserver":
+				start.NetworkServer = true
+				start.DeviceTemplateConverter = true
+				start.QRCodeGenerator = true
+			case "as", "applicationserver":
+				start.ApplicationServer = true
+				start.DeviceTemplateConverter = true
+				start.QRCodeGenerator = true
+			case "js", "joinserver":
+				start.JoinServer = true
+				start.DeviceTemplateConverter = true
+				start.QRCodeGenerator = true
+			case "console":
+				start.Console = true
+			case "gcs":
+				start.GatewayConfigurationServer = true
+			case "dtc":
+				start.DeviceTemplateConverter = true
+			case "qrg":
+				start.QRCodeGenerator = true
+			case "all":
+				start.IdentityServer = true
+				start.GatewayServer = true
+				start.NetworkServer = true
+				start.ApplicationServer = true
+				start.JoinServer = true
+				start.Console = true
+				start.GatewayConfigurationServer = true
+				start.DeviceTemplateConverter = true
+				start.QRCodeGenerator = true
+			default:
+				return errUnknownComponent.WithAttributes("component", arg)
 			}
+		}
 
-			logger.Info("Setting up core component")
+		logger.Info("Setting up core component")
 
-			var rootRedirect web.Registerer
+		var rootRedirect web.Registerer
 
-			var componentOptions []component.Option
+		var componentOptions []component.Option
 
-			c, err := component.New(logger, &component.Config{ServiceBase: config.ServiceBase}, componentOptions...)
-			if err != nil {
-				return shared.ErrInitializeBaseComponent.WithCause(err)
-			}
+		c, err := component.New(logger, &component.Config{ServiceBase: config.ServiceBase}, componentOptions...)
+		if err != nil {
+			return shared.ErrInitializeBaseComponent.WithCause(err)
+		}
+
+		c.RegisterGRPC(events_grpc.NewEventsServer(c.Context(), events.DefaultPubSub()))
+		c.RegisterGRPC(component.NewConfigurationServer(c))
 
-			c.RegisterGRPC(events_grpc.NewEventsServer(c.Context(), events.DefaultPubSub()))
-			c.RegisterGRPC(component.NewConfigurationServer(c))
+		host, err := os.Hostname()
+		if err != nil {
+			return err
+		}
 
-			host, err := os.Hostname()
+		if start.IdentityServer || startDefault {
+			logger.Info("Setting up Identity Server")
+			is, err := identityserver.New(c, &config.IS)
 			if err != nil {
-				return err
+				return shared.ErrInitializeIdentityServer.WithCause(err)
 			}
-
-			if start.IdentityServer || startDefault {
-				logger.Info("Setting up Identity Server")
-				is, err := identityserver.New(c, &config.IS)
-				if err != nil {
-					return shared.ErrInitializeIdentityServer.WithCause(err)
-				}
+			if config.Cache.Service == "redis" {
 				is.SetRedisCache(redis.New(&redis.Config{
 					Redis:     config.Redis,
```

Comment:
Shouldn't this then be `config.Cache.Redis` instead of `config.Redis`?

Reply:
Indeed. Stupid me copied the wrong code while rebasing.